go.mod

```diff
@@ -17,6 +18,7 @@ require (
 	github.com/google/go-github/v61 v61.0.0
 	github.com/kelseyhightower/envconfig v1.4.0
 	github.com/prometheus/client_golang v1.19.0
+	github.com/snabb/httpreaderat v1.0.1
```

Comment:
@jonjohnsonjr has toyed with a similar thing, which might work and be simpler: https://github.com/jonjohnsonjr/targz/blob/main/ranger/ranger.go

Reply:
I stole this impl from him, you beat me to giving him credit!